src/controllers/organisationJobs.ts

```diff
@@ -0,0 +1,161 @@
+import moment from 'moment-timezone';
+import { CreateOrganisationJobSchema } from '../database/models';
+import { OrganisationJobsRepository, OrganisationsRepository } from '../database';
+import { asyncHandler } from '../middleware';
+import { checkOrganisationOwner, cleanseData, ErrorResponse } from '../utils';
+
+export class OrganisationJobsController {
+    constructor(
+        private readonly organisationJobsRepository: OrganisationJobsRepository,
+        private readonly organisationsRepository: OrganisationsRepository,
+    ) {
+        this.organisationJobsRepository = organisationJobsRepository;
+        this.organisationsRepository = organisationsRepository;
+    }
+
+    /**
+     * @desc    Get all jobs for a organisation
+     * @route   GET /api/organisations/:organisationId/jobs
+     * @access  Public
+     */
+    getOrganisationJobs = asyncHandler(async (req, res, next) => {
+        if (req.params.organisation_id) {
+            const organisationId: string = req.params.organisation_id;
+
+            // return 404 error response if organisation not found
+            await this.organisationsRepository.getOrganisationById(organisationId);
+
+            const jobs = await this.organisationJobsRepository.getAllJobsForOrganisation(organisationId);
+            return res.status(200).json({
+                success: true,
+                data: jobs,
+            });
+        }
+
+        return next(new ErrorResponse('Invalid route', 404));
+    });
+
+    /**
+     * @desc    Get single organisation job
+     * @route   GET /api/organisation-jobs/:organisationJobId
+     * @access  Public
+     */
+    getSingleOrganisationJob = asyncHandler(async (req, res, next) => {
+        const organisationJobId = req.params.organisationJobId;
```

Comment:
Can make use of JS destructuring here:
```suggestion
        const { organisationJobId } = req.params;
```